ipatests/test_webui/test_subid.py

```diff
@@ -142,17 +141,7 @@ def test_subid_range_deletion_not_allowed(self):
         """
         self.init_app()
         self.navigate_to_entity('subid', facet='search')
-        self.facet_button_click('add')
-        self.select_combobox('ipaowner', 'admin')
-        self.dialog_button_click('add')
-        self.wait(0.3)
-        self.assert_no_error_dialog()
-        self.get_field_checked('ipauniqueid')
+        admin_uid = self.get_record_pkey("admin", "ipaowner",
+                                         table_name="ipauniqueid")
         with pytest.raises(NoSuchElementException):
-            self.facet_button_click('remove')
-        buttons_s = 'div.control-buttons button'
-        facet_buttons = self.find(buttons_s, By.CSS_SELECTOR,
-                                  many=True, strict=True)
-        assert len(facet_buttons) == 2
-        assert facet_buttons[0].get_attribute('name') in ['add', 'refresh']
-        assert facet_buttons[1].get_attribute('name') in ['add', 'refresh']
+            self.delete_record(admin_uid, table_name="ipauniqueid")
```

Comment:
Take this with a grain of salt, I don't know webUI testing well, but how is this testing the non-existence of a button?

I assume that delete_record() raises the NoSuchElementException? Does that mean that there is no related button, or that the record can't be deleted, or it doesn't exist?

Reply:
Yes, the selenium driver returns a NoSuchElementException when it can't find a button:
```
<ExceptionInfo NoSuchElementException('Unable to locate element: .facet-controls button[name=remove]', None, ['RemoteError@chrome://r...ntent/shared/webdriver/Errors.jsm:404:5', 'element.find/</<@chrome://remote/content/marionette/element.js:291:16', '']) tblen=10>
```
I'm adding a more precise check on the exception to make sure the test complains about the missing button and not something else.